File: 20260509/exos/05_while.py
# ============================================================
# Project Zomboid — générateur bricolé à la main
# ============================================================
# Tu tires une poignée de levier jusqu’à ce que la jauge d’électricité dépasse
# un minimum pour rallumer un radiateur. Tu augmentes la jauge de +18 à chaque essai.
#
# Lance : python3 05_while.py
# ============================================================

electricite = 9

# Répète des tours tant que electricite est strictement inférieure à 60.
# À chaque tour : ajoute 18 à electricite.
#
# départ : 9
# tour 1 : 9 + 18 = 27   (27 < 60 → continue)
# tour 2 : 27 + 18 = 45  (45 < 60 → continue)
# tour 3 : 45 + 18 = 63  (63 < 60 ? non → arrêt)
#
# Résultat attendu : electricite == 63
#
# Indice : while ... : puis += dans la boucle

# À toi :
while electricite < 60:
    electricite += 18

# --- Vérification (ne pas modifier) ---
assert electricite == 63, (
    f"Obtenu : {electricite} — après trois ajouts de 18 depuis 9, on attend 63."
)
print("✅ Correct !")

File: 20260722/exos/05_while.lua
-- ============================================================
-- Hearts of Iron IV — moral qui remonte
-- ============================================================
-- Le moral d'une division part à 10. Tant qu'il est strictement
-- inférieur à 50, il gagne 10. Quelle valeur finale ?
--
-- Lance : lua 05_while.lua
-- ============================================================

moral = 10

-- Tant que moral < 50, ajoute 10.
--
-- 10 → 20 → 30 → 40 → 50  puis stop (50 < 50 ? non)
--
-- Résultat attendu : moral == 50
--
-- Indice : while ... do ... end
--          et moral = moral + 10

-- À toi :
while moral < 50 do
    moral = moral + 10
end
-- --- Vérification (ne pas modifier) ---
assert(moral == 50, "Obtenu : " .. moral .. ", attendu : 50")
print("✅ Correct !")
